internal/pkg/deploy/cloudformation/env.go

```diff
@@ -238,6 +254,9 @@ func (cf CloudFormation) toUploadedStack(artifactBucketARN string, stackConfig S
 }
 
 func (cf CloudFormation) waitAndDescribeStack(stackName string) (*cloudformation.StackDescription, error) {
+	if cf.cachedDeployedStack != nil {
+		return cf.cachedDeployedStack, nil
```

Comment:
is there a scenario where caching can be problematic? are there any dependent functions that you think would want the latest value. For example, if we cache the `StackStatus` will never be updated

Reply:
> are there any dependent functions that you think would want the latest value

There are currently two dependent calls, one is getting the old parameters, the other is getting for force update ID. Between these two, I think it's safe to assume that no matter which is the first to invoke this function, the second call follows immediately such that the cached stack is the latest stack.

I do see your concern though! I felt like retrieving old parameters and force update ID do not warrant two API calls. I can refactor the functions such that we don't need to cache while still maintaining just one API call.